src/Core/Streaming/Watermark.h

```diff
@@ -9,5 +9,48 @@ namespace Streaming
 /// The 1970-01-01 00:00:00 is Int64 value 0, earlier times are negative, so we need to use signed Int64 as watermark
 constexpr Int64 INVALID_WATERMARK = std::numeric_limits<Int64>::min();
 constexpr Int64 TIMEOUT_WATERMARK = std::numeric_limits<Int64>::max();
+
+
+enum class WatermarkStrategy
+{
+    Unknown,
+
+    /// No watermark
+    None,
+
+    /// Watermark is generated based on each event, this is used for `EMIT AFTER WATERMARK WITHOUT DELAY`
+    /// - Not allow time skew
+    Ascending,
+
+    /// Watermark is generated based on last event with a delay interval, this is used for `EMIT AFTER WATERMARK WITH DELAY <delay_interval>`:
+    /// - Allow time skew in a certain range `[<max_event_ts - delay_interval>, <max_event_ts>]`)
+    BoundedOutOfOrderness,
```

Comment:
We may need revise the names since we don't understand what they means.

Reply:
Just `Delay` which matches the emit clause is more readable / understandable, OutOfOrder is the result, not the strategy itself